File: Java Version/Encryption.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.File;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;
import java.util.Base64;

public class Encryption {
    private static final int ByteFF = 0xFF;
    public int globalkey;

    public String encodeToBase64(String encyptedString) {
        return Base64.getEncoder().encodeToString(encyptedString.getBytes());
    }

    public byte[] decodeToBase64(String decodedString) {
        return Base64.getDecoder().decode(decodedString);
    }

    public String convertToString(byte[] arrayOfStrings) {
        return new String(arrayOfStrings);
    }

    public String encryptDecrypt(String input) {
        List<Byte> inputBytes = new ArrayList<Byte>();
        for (byte b : input.getBytes()) {
            inputBytes.add(b); // 121-65-89
        }

        List<Byte> result = new ArrayList<Byte>(inputBytes.size());
        for (byte bb : inputBytes) {
            result.add((byte) (bb ^ (globalkey & ByteFF))); // XOR
        }

        StringBuilder converted = new StringBuilder();
        for (byte b : result) {
            converted.append((char) b);
        }
        return converted.toString();
    }

    // DITO LALAGAY YUNG PROMPT FOR USER TO EDIT KEY
    Encryption() {
        this.setKey(globalkey);
    }

    public int getKey() {
        return this.globalkey;
    }

    public void setKey(int key) {
        this.globalkey = key;
    }

    public void saveKey() {
        // Save the key here
        try (BufferedWriter outFile = new BufferedWriter(new FileWriter(Variables.keyFilePath));) {
            outFile.write(Integer.toString(globalkey));
        } catch( IOException e) {
            System.out.println("Error opening/writing to file: " + e.getMessage());
        }
    }

    public void retrieveKey() {
        try (BufferedReader inFile = new BufferedReader(new FileReader(Variables.keyFilePath));) {
            String key = inFile.readLine();
            setKey(Integer.parseInt(key));
        } catch( IOException e) {
            System.out.println("Error opening/writing to file: " + e.getMessage());
        }
    }
}

File: Python Version/Encryption.py
import base64
from main import Variables

globalkey = 0
ByteFF = 0xFF
RAND_SEED = 42
CHAR_NUM = 26
NUM_NUM = 10


class Encryption:
    def __init__(self):
        self.globalkey = None


def getKey(self):
    return self.globalkey


def setKey(key):
    global globalkey
    globalkey = key


def encrypt(plain_text):
    encrypted = []
    for char in plain_text:
        if char.isalpha():
            if char.islower():
                encrypted_char = chr((ord(char) - ord('a') + RAND_SEED) % CHAR_NUM + ord('a'))
            else:
                encrypted_char = chr((ord(char) - ord('A') + RAND_SEED) % CHAR_NUM + ord('A'))
        elif char.isdigit():
            encrypted_char = chr((ord(char) - ord('0') + RAND_SEED) % NUM_NUM + ord('0'))
        else:
            encrypted_char = char
        encrypted.append(encrypted_char)
    return ','.join(encrypted)


def decrypt(encrypted_text):
    decrypted = []
    encrypted_chars = encrypted_text.split(',')
    for char in encrypted_chars:
        if char.isalpha():
            if char.islower():
                decrypted_char = chr((ord(char) - ord('a') - RAND_SEED + CHAR_NUM) % CHAR_NUM + ord('a'))
            else:
                decrypted_char = chr((ord(char) - ord('A') - RAND_SEED + CHAR_NUM) % CHAR_NUM + ord('A'))
        elif char.isdigit():
            decrypted_char = chr((ord(char) - ord('0') - RAND_SEED + NUM_NUM) % NUM_NUM + ord('0'))
        else:
            decrypted_char = char
        decrypted.append(decrypted_char)
    return ''.join(decrypted)


def convertToString(arrayofStrings):
    result = ''.join(arrayofStrings)
    return result


def encodeToBase64(encryptedStrings):
    return base64.b64encode(encryptedStrings.encode())


def decodeToBase64(decryptedStrings):
    return base64.b64decode(decryptedStrings)


def saveKey():
    try:
        with open(Variables.KEY_FILE, 'w') as outFile:
            outFile.write(str(globalkey))
    except IOError as e:
        print("Error opening/writing to file: ", e)


def retrieveKey():
    encryption = Encryption()
    try:
        with open(Variables.KEY_FILE, 'r') as inFile:
            key = int(inFile.readline())
            setKey(key)
    except IOError as e:
        print("Error opening/reading the file: ", e)
